### ISSUE TITLE
[HASKELL] : Error on type object

### ISSUE THREAD

**Issue by joe9** [swagger-api/swagger-codegen#3396]
##### Description

When there is a type of object in the swagger data format file, the haskell generator is failing with an exception. The python generator works fine.

##### Swagger-codegen version

0d9878373c0ab24abfdbd38f12b0509ee32e34fb - latest master release

##### Swagger declaration file content or url

[https://gist.github.com/b051e2e05d1584948bd67a645ebec2ae is the swagger data format file](https://gist.github.com/b051e2e05d1584948bd67a645ebec2ae)

##### Command line used for generation

```
JAVA_HOME=/usr/lib/jvm//icedtea-bin-8 PATH="${JAVA_HOME}/bin:$PATH" java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -i ESASwaggerSchema.json -l haskell -o /tmp/haskell-client
[main] INFO io.swagger.parser.Swagger20Parser - reading from ESASwaggerSchema.json
Exception in thread "main" java.lang.RuntimeException: Could not process model 'AuthenticationMessage'.Please make sure that your schema is correct!
        at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:290)
        at io.swagger.codegen.cmd.Generate.run(Generate.java:223)
        at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:36)
Caused by: java.lang.ClassCastException: io.swagger.models.ComposedModel cannot be cast to io.swagger.models.ModelImpl
        at io.swagger.codegen.languages.HaskellServantCodegen.fromModel(HaskellServantCodegen.java:474)
        at io.swagger.codegen.DefaultGenerator.processModels(DefaultGenerator.java:868)
        at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:283)
        ... 2 more
```

##### Steps to reproduce

```
JAVA_HOME=/usr/lib/jvm//icedtea-bin-8 PATH="${JAVA_HOME}/bin:$PATH" java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -i ESASwaggerSchema.json -l haskell -o /tmp/haskell-client
[main] INFO io.swagger.parser.Swagger20Parser - reading from ESASwaggerSchema.json
Exception in thread "main" java.lang.RuntimeException: Could not process model 'AuthenticationMessage'.Please make sure that your schema is correct!
        at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:290)
        at io.swagger.codegen.cmd.Generate.run(Generate.java:223)
        at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:36)
Caused by: java.lang.ClassCastException: io.swagger.models.ComposedModel cannot be cast to io.swagger.models.ModelImpl
        at io.swagger.codegen.languages.HaskellServantCodegen.fromModel(HaskellServantCodegen.java:474)
        at io.swagger.codegen.DefaultGenerator.processModels(DefaultGenerator.java:868)
        at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:283)
        ... 2 more
```

##### Suggest a Fix

The above seems to be happening where the "type: object" is in the definition.

@algas @gibiansky @wing328

**Comment by wing328** [comment]
@joe9 thanks for reporting the issue, which has been fixed by #3647. Please pull the latest master to give it a try.

**Comment by joe9** [comment]
Thanks @wing328 , it worked like a charm.
